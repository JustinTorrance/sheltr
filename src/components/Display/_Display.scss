.pets-results {
  width: 100%;
  position: absolute;
  top: 340px;
  display: flex;
  justify-content: center;
}

.pets-grid {
  color: white;
  position: absolute;
  margin-top: 100px;
  display: grid;
  grid-template-columns: repeat(5, 250px);
  width: 100%;
  grid-gap: 30px;
  justify-content: center;
}

.shelter-name {
  color: white;
  font-size: 3rem;
  text-shadow:
      -1px -1px 0 black,  
      1px -1px 0 black,
      -1px 1px 0 black,
      1px 1px 0 black;
}

.pet-wrapper {
  border: solid 2px white;
  border-radius: 20px;
  height: 400px;
  text-align: center;
  font-size: 1.4rem;
  text-shadow:
    -1px -1px 0 black,  
    1px -1px 0 black,
    -1px 1px 0 black,
    1px 1px 0 black;

  p {
    margin-top: 10px;
  }

  .pet-name {
    font-size: 1.3rem;
  }
}

.pet-photo {
  height: 200px;
  width: 246px;
  border-radius: 20px;
}

.fa-heart {
  color: white;
  position: relative;
  left: 85px;
    &:hover {
      color: red;
      cursor: pointer;
    }
}
