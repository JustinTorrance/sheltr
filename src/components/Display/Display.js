import React from 'react'
import { connect } from 'react-redux'
import './_Display.scss'
import { addFavorite, removeFavorite } from '../../actions'
import Card from '../Card/Card.js'
import { Link } from 'react-router-dom'

export const Display = (props) => {

  return (
    <section className='pets-results'>
      <Link to='/favorites'>
        <button>View Favorites</button>
      </Link>
      <h3 className='shelter-name'>Pets Available At This Shelter</h3>
      <section className='pets-grid'>
        {
          props.petsFromShelter.map(pet => {
            return (
              <Card {...pet} />
            )            
          })          
        }
      </section>
    </section>
  )
}

export const mapStateToProps = (state) => ({
  petsFromShelter: state.petsFromShelter,
  favorites: state.favorites
})

export const mapDispatchToProps = (dispatch) => ({
  addFavorite: (pet) => dispatch(addFavorite(pet)),
  removeFavorite: (pet) => dispatch(removeFavorite(pet))
})

export default connect(mapStateToProps, mapDispatchToProps)(Display)