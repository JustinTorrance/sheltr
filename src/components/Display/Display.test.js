import { mapStateToProps } from './Display'
import Display from './Display'
import React from 'react';


describe('Display', () => {
  it('matches the snapshot', () => {
    const wrapper = <Display />
    expect(wrapper).toMatchSnapshot()
  })

  describe('mapStateToProps', () => {
    it('should return an array from pets from a shelter', () => {
      const mockState = {
        petsFromShelter: [{name: 'fluffy'}]
      }
      const expected = {
        petsFromShelter: [{name: 'fluffy'}] 
      }
      const mappedProps = mapStateToProps(mockState)

      expect(mappedProps).toEqual(expected)
    })
  })

  describe('mapDispatchToProps', () => {
    it('calls dispatch with a addFavorite thunk when handeSubmit is called', () => {
      const mockDispatch = jest.fn()
      const thunkToDispatch = addFavorite(e)
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.toggleFavorite(e)
      expect(mockDispatch).toHaveBeenCalledWith(thunkToDispatch)
    })
    it('calls dispatch with a removeFavorite thunk when handeSubmit is called', () => {
      const mockDispatch = jest.fn()
      const thunkToDispatch = removeFavorite(e)
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.toggleFavorite(e)
      expect(mockDispatch).toHaveBeenCalledWith(thunkToDispatch)
    })
  })
})