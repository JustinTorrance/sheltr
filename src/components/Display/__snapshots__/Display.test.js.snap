// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display matches the snapshot 1`] = `<Connect(Display) />`;
